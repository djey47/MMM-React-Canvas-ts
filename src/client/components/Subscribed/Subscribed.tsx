import { ComponentType } from 'react';
import useWithNotifications from '../../hooks/with-notifications/with-notifications';

interface OwnProps {
  prop1: string;
}

interface NotificationData {
  data_NOTIF?: object;
}

const SubscribedSample: ComponentType<OwnProps> = (
  props: OwnProps
) => {
  const notificationData = useWithNotifications(['NOTIF']) as NotificationData;
  const data_NOTIF = notificationData?.data_NOTIF;

  return (
    <div className="subscribed">
      <p className="subscribed__text">Subscribed component to 'NOTIF' event</p>
      <p className="subscribed__props">
        Own props: {JSON.stringify(props)}
      </p>
      <p className="subscribed__value">
        Last received value: {JSON.stringify(data_NOTIF)}
      </p>
    </div>
  );
};

export default SubscribedSample;
