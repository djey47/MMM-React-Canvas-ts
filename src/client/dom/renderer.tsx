import ReactDOM from 'react-dom';

import MainSample from '../components/Main/MainSample';

/**
 * MagicMirror
 * HTML renderer
 */

/**
 * @returns HTML for main wrapper
 */
export const renderWrapper = (wrapperId: string): HTMLDivElement => {
  const wrapper = document.createElement('div');
  wrapper.id = wrapperId;
  wrapper.className = wrapperId;

  return wrapper;
};

/**
 * REACT gateway helper
 * @return Mounted component
 */
export const renderMainComponent = (wrapperId: string): void => {
  const rootId = document.getElementById(wrapperId);
  if (!rootId) {
    Log.error(`** Could not find root div with id: ${wrapperId}! Aborting.`);
    return;
  }

  ReactDOM.render(
    <MainSample />,
    rootId
  );
};
