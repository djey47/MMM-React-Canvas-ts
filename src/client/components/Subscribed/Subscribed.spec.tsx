// Must be located BEFORE the imports
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const mockUseWithNotifications = jest.fn();

import renderer from 'react-test-renderer';
import Subscribed from './Subscribed';

jest.mock('../../hooks/with-notifications/with-notifications', () => mockUseWithNotifications);

describe('Subscribed component', () => {
  beforeEach(() => {
    mockUseWithNotifications.mockReset();
  });

  it('should render correctly without notif data', () => {
    const tree = renderer.create(<Subscribed prop1="val1" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockUseWithNotifications).toHaveBeenCalledWith(['NOTIF']);
  });

  it('should render correctly with notif data', () => {
    // given
    mockUseWithNotifications.mockReturnValue({ data_NOTIF: { key: 'value' }});

    // when-then
    const tree = renderer
      .create(<Subscribed prop1="val1" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
