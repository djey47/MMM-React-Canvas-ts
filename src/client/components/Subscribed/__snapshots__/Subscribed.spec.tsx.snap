// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribed component should render correctly with notif data 1`] = `
<div
  className="subscribed"
>
  <p
    className="subscribed__text"
  >
    Subscribed component to 'NOTIF' event
  </p>
  <p
    className="subscribed__props"
  >
    Own props: 
    {"prop1":"val1"}
  </p>
  <p
    className="subscribed__value"
  >
    Last received value: 
    {"key":"value"}
  </p>
</div>
`;

exports[`Subscribed component should render correctly without notif data 1`] = `
<div
  className="subscribed"
>
  <p
    className="subscribed__text"
  >
    Subscribed component to 'NOTIF' event
  </p>
  <p
    className="subscribed__props"
  >
    Own props: 
    {"prop1":"val1"}
  </p>
  <p
    className="subscribed__value"
  >
    Last received value: 
  </p>
</div>
`;
